pkg/mixin/feed/generate.go

```diff
@@ -0,0 +1,168 @@
+package feed
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"sort"
+	"time"
+
+	"github.com/cbroglie/mustache"
+	"github.com/deislabs/porter/pkg/context"
+	"github.com/pkg/errors"
+)
+
+type GenerateOptions struct {
+	SearchDirectory string
+	AtomFile        string
+	TemplateFile    string
+}
+
+func (o *GenerateOptions) Validate(c *context.Context) error {
+	err := o.ValidateSearchDirectory(c)
+	if err != nil {
+		return err
+	}
+
+	return o.ValidateTemplateFile(c)
+}
+
+func (o *GenerateOptions) ValidateSearchDirectory(cxt *context.Context) error {
+	if o.SearchDirectory == "" {
+		wd, err := os.Getwd()
+		if err != nil {
+			return errors.Wrap(err, "could not get current working directory")
+		}
+
+		o.SearchDirectory = wd
+	}
+
+	if _, err := cxt.FileSystem.Stat(o.SearchDirectory); err != nil {
+		return errors.Wrapf(err, "invalid --directory %s", o.SearchDirectory)
```

Comment:
Q: should we print `invalid --dir` or `invalid -d` to represent the flag strings supported?  Or perhaps just `invalid directory` for a more general error msg?

Reply:
oops, yes thanks it should be `--dir`